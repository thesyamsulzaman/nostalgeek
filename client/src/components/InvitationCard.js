import React, { Component } from "react";
import PropTypes from "prop-types";
import { Card, Icon, Image } from "semantic-ui-react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";

import LikeButton from "../components/LikeButton";

class InvitationCard extends Component {
  render() {
    const {
      invitation: { id, title, body, likeCount, commentCount, imageUrl },
      actions = null,
    } = this.props;

    return (
      <Card style={{ width: "100%" }}>
        <Image
          src={imageUrl}
          size="medium"
          style={{ width: "100%", height: "150px", objectFit: "cover" }}
        />
        <Card.Content>
          <Card.Header as={Link} to={`/invitations/${id}`}>
            {title}
          </Card.Header>
          <Card.Description>{body}</Card.Description>
        </Card.Content>

        <Card.Content style={{ display: "flex", alignItems: "center" }}>
          <div style={{ display: "flex", alignItems: "center", flex: 1 }}>
            <div style={{ marginRight: ".5em" }}>
              <LikeButton key={id} likeCount={likeCount} invitationId={id} />
            </div>

            <div>
              {commentCount} <Icon name="chat" />
            </div>
          </div>

          {actions}
        </Card.Content>
      </Card>
    );
  }
}

InvitationCard.propTypes = {
  invitation: PropTypes.shape({
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    likeCount: PropTypes.number.isRequired,
    commentCount: PropTypes.number.isRequired,
  }),
  loading: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  loading: state.ui.loading,
});

const mapActionsToProps = {};

export default connect(mapStateToProps, mapActionsToProps)(InvitationCard);
