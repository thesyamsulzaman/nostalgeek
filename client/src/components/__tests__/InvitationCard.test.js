import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import InvitationCard from '../InvitationCard';

const mockStore = configureStore([]);

describe('Testing Redux integrated Component', () => {
  let store;
  let component;

  // beforeEach(() => {
  //   store = mockStore({
  //     invitations: {},
  //     ui: {},
  //     user: {},
  //   });

  //   component = renderer.create(
  //     <Provider store={store}>
  //       <Router>
  //         <InvitationCard
  //           invitation={{ title: "", body: "", likeCount: 0, commentCount: 0 }}
  //           user={{}}
  //           likeInvitation={() => {}}
  //           unlikeInvitation={() => {}}
  //         />
  //       </Router>
  //     </Provider>
  //   );
  // });

  it('Should render the unlike invitation', () => {
    expect(1).toEqual(1);
  });
  // it("Should dispatch the like action", () => {});
});
