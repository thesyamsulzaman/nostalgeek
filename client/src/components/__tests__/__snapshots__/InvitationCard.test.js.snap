// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Redux integrated Component Should render the unlike invitation 1`] = `
<div
  className="ui card"
  onClick={[Function]}
  style={
    Object {
      "width": "100%",
    }
  }
>
  <img
    className="ui medium image"
    style={
      Object {
        "height": "150px",
        "objectFit": "cover",
        "width": "100%",
      }
    }
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      
    </div>
    <div
      className="description"
    >
      
    </div>
  </div>
  <div
    className="content"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
        }
      }
    >
      <div
        style={
          Object {
            "marginRight": ".5em",
          }
        }
      >
        <div
          className="ui left labeled button"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          <a
            className="ui right pointing basic label"
            onClick={[Function]}
          >
            0
          </a>
          <a
            className="ui button"
            href="/user/login"
            onClick={[Function]}
            role="button"
          >
            <i
              aria-hidden="true"
              className="thumbs up outline icon"
              onClick={[Function]}
            />
            Like
          </a>
        </div>
      </div>
      <div>
        0
         
        <i
          aria-hidden="true"
          className="chat icon"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
